### PR TITLE
ARC64: Update/Add ARC64 related compiler sections

### PR DESCRIPTION
**Pull request by claziss**
Update ARC64 compiler options.

### CODE REVIEW

File: doc/baremetal/gcc-mcpu.rst [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+The GCC toolchain has support for three ARC CPU families, namely
+
+* ARCv1 CPU family made of ARC6xx, and ARC7xx processors;
+* ARCv2 CPU family made of ARCEM, and ARCHS procesors;
+* ARCv3 CPU family made of ARC3:HS5x, and ARC64:HS6x processors.
+
+All the above processors are 32bit RISC processors. Notable exception
+is ARC64:HS6x processors which is a 64bit RISC machine.
+
+The next table depicts which GCC driver shoudl be used depending on
```
Comment (shahab-vahedi) [inline doc/baremetal/gcc-mcpu.rst]: `should`

File: doc/baremetal/gcc-mcpu.rst [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+ARC Toolchain Variants
+======================
+
+The GCC toolchain has support for three ARC CPU families, namely
+
+* ARCv1 CPU family made of ARC6xx, and ARC7xx processors;
+* ARCv2 CPU family made of ARCEM, and ARCHS procesors;
+* ARCv3 CPU family made of ARC3:HS5x, and ARC64:HS6x processors.
+
+All the above processors are 32bit RISC processors. Notable exception
```
Comment (shahab-vahedi) [inline doc/baremetal/gcc-mcpu.rst]: `Most of the above`

File: doc/baremetal/gcc-mcpu.rst [hunk 3]
```diff
@@ -161,4 +186,62 @@ values for ARC 600 and ARC 700.
...
+   ======== ======== ======== ========
+    -mcpu    -msimd   -m128    -mll64
+   ======== ======== ======== ========
+   hs5x       Y       N.A.      N
+   hs58       Y       N.A.      Y
+   hs6x       Y       N         N.A.
+   hs68       Y       Y         N.A.
+   ======== ======== ======== ========
+
+In baremetal, we provide a number of prebuild multilib libraries. These are:
```
Comment (shahab-vahedi) [inline doc/baremetal/gcc-mcpu.rst]: `pre-built`

**Review by shahab-vahedi — APPROVED** [review]

File: doc/baremetal/gcc-mcpu.rst [hunk 4]
```diff
@@ -1,7 +1,32 @@
...
-Understanding GCC -mcpu option
-==============================
+ARC Toolchain Variants
+======================
+
+The GCC toolchain has support for three ARC CPU families, namely
+
+* ARCv1 CPU family made of ARC6xx, and ARC7xx processors;
+* ARCv2 CPU family made of ARCEM, and ARCHS procesors;
+* ARCv3 CPU family made of ARC3:HS5x, and ARC64:HS6x processors.
```
Comment (abrodkin) [inline doc/baremetal/gcc-mcpu.rst]: `ARC3:HS5x` -> `ARC32:HS5x`

File: doc/baremetal/gcc-mcpu.rst [hunk 5]
```diff
@@ -1,7 +1,32 @@
 .. index:: mcpu, compiler
 
-Understanding GCC -mcpu option
-==============================
+ARC Toolchain Variants
+======================
+
+The GCC toolchain has support for three ARC CPU families, namely
+
+* ARCv1 CPU family made of ARC6xx, and ARC7xx processors;
```
Comment (abrodkin) [inline doc/baremetal/gcc-mcpu.rst]: ARCompact, also known as ARCv1

File: doc/baremetal/gcc-mcpu.rst [hunk 6]
```diff
@@ -161,4 +186,62 @@ values for ARC 600 and ARC 700.
...
+
+The next ``-mcpu`` options are recognized by the ARC64 GCC compiler.
+
+.. table:: -mcpu variants for ARCv3
+
+   ======== ======== ======== ========
+    -mcpu    -msimd   -m128    -mll64
+   ======== ======== ======== ========
+   hs5x       Y       N.A.      N
+   hs58       Y       N.A.      Y
```
Comment (abrodkin) [inline doc/baremetal/gcc-mcpu.rst:201]: Doesn't `hs58` include "advanced mpy9 SIMD instructions" compared to `hs5x`?

**Review by abrodkin — COMMENTED** [review]
See a couple of tiny fixes.

File: doc/baremetal/gcc-mcpu.rst (hunk 6 above)
Comment (claziss) [reply doc/baremetal/gcc-mcpu.rst:201]: That should be under `-msimd` option. Do you suggest to have `hs5x`  to be without `-msimd` option?

Comment (abrodkin) [reply doc/baremetal/gcc-mcpu.rst:201]: Well, I somehow understood that incorrectly. And so I suggest to keep settings as they are now. Sorry for the noise.

File: doc/baremetal/gcc-mcpu.rst (hunk 4 above)
Comment (abrodkin) [reply doc/baremetal/gcc-mcpu.rst]: @claziss care to fix that spello?

**Review by abrodkin — CHANGES_REQUESTED** [review]

**Comment by kolerov** [comment]
@abrodkin 
I think it's better to push this pull-request at first:

https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/pull/516

Documentation will not build successfully without.

**Comment by abrodkin** [comment]
@kolerov sure, please go ahead and merge both of them in the right order.

**Comment by kolerov** [comment]
This commit was pushed to `arc-dev` with small fixes: https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/commit/7619f73e65fb36e9f989cd5ec44311cbeb680524.

@abrodkin I've pushed it to `arc-dev` because it already contains change for documentation in a long history beyond `arc-releases`. It's necessary to push accumulated changes from `arc-dev` to `arc-releases`. Before that we cannot modify the documentation on top of `arc-releases` branche since it contains outdated script for building HTML pages.

**Pull request closed without merging**